### PR TITLE
Generate python3 docs

### PR DESCRIPTION
**Pull request by booxter**
This patch also updates generate.sh to new python.org docs archives layout.

### CODE REVIEW

**Comment by booxter** [comment]
I was also looking at how to properly link the documents to /usr/share/ but haven't figured it out myself yet. If someone has an idea or a direction to pursue, please share.

**Comment by FRidh** [comment]
It's good this got some attention, thanks!

**Comment by FRidh** [comment]
> I was also looking at how to properly link the documents to /usr/share/ but haven't figured it out myself yet.

In Nixpkgs we don't use `/usr`. It's contents goes directly in `/`.

**Comment by booxter** [comment]
@FRidh right, I missed that, still new to the platform. Looks like I can still refer to the docs through my ```~/.nix-profile```, so that's good enough.

**Comment by booxter** [comment]
Anything that holds this PR?

File: pkgs/development/interpreters/python/cpython/docs/2.7-html.nix [hunk 1]
```diff
@@ -3,10 +3,10 @@
 { stdenv, fetchurl, lib }:
 
 stdenv.mkDerivation rec {
-  name = "python27-docs-html-2.7.3";
+  name = "python27-docs-html-2.7.16";
```
Comment (FRidh) [inline pkgs/development/interpreters/python/cpython/docs/2.7-html.nix]: Would be good to pick up version info from the `python27.version` attribute

File: pkgs/development/interpreters/python/cpython/docs/2.7-html.nix [hunk 2]
```diff
@@ -3,10 +3,10 @@
...
   src = fetchurl {
-    url = http://docs.python.org/ftp/python/doc/2.7.3/python-2.7.3-docs-html.tar.bz2;
-    sha256 = "1hg92n0mzl9w6j33b2h0bf2vy6fsxnpxfdc3qw760vcm0y00155j";
+    url = http://docs.python.org/ftp/python/doc/2.7.16/python-2.7.16-docs-html.tar.bz2;
```
Comment (FRidh) [inline pkgs/development/interpreters/python/cpython/docs/2.7-html.nix]: and then use it here

**Review by FRidh — COMMENTED** [review]
Will merge as is, even though some parts are already outdated, because it's an improvement anyway. Thanks!

**Pull request merged**